### PR TITLE
lightning/restore: support ingset multi ssts for same range

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Ingest multi SST files (different CF) together to gain better consistency. This can resolve the problem that lighting import or BR restore may cause TiFlash panic.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has interface methods change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - No Release Note

### CODE REVIEW

File: pkg/restore/import.go [hunk 1]
```diff
@@ -155,16 +177,32 @@ func (ic *importClient) GetImportClient(
 	return client, errors.Trace(err)
 }
 
+func (ic *importClient) SupportMultiIngest(ctx context.Context, stores []uint64) (bool, error) {
```
Comment (Little-Wallace) [inline pkg/restore/import.go:180]: Why storeId is an array?

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: pkg/lightning/backend/local/local.go [hunk 2]
```diff
@@ -868,9 +871,39 @@ func NewLocalBackend(
...
+	for _, s := range stores {
+		client, err := local.getImportClient(ctx, s.Id)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		_, err = client.MultiIngest(ctx, &sst.MultiIngestRequest{})
+		if err != nil {
+			if s, ok := status.FromError(err); ok {
+				if s.Code() == codes.Unimplemented {
+					local.supportMultiIngest = false
```
Comment (solotzg) [inline pkg/lightning/backend/local/local.go:895]: Does tiflash need to implement such interfaces? @Little-Wallace

Comment (Little-Wallace) [reply pkg/lightning/backend/local/local.go:895]: No

Comment (solotzg) [reply pkg/lightning/backend/local/local.go:895]: It seems that `supportMultiIngest` is true only when all stores have implemented related interfaces.

Comment (glorv) [reply pkg/lightning/backend/local/local.go:901]: MultiIngest is only called by region leader, so tiflash needn't implement this interface. We should skip check tiflash here

File: pkg/restore/client.go [hunk 3]
```diff
@@ -541,6 +541,38 @@ func (rc *Client) setSpeedLimit(ctx context.Context) error {
...
+	}
+	idx := 1
+	for idx < len(files) {
+		if !isFilesBelongToSameRange(files[idx-1].Name, files[idx].Name) {
+			break
+		}
+		idx++
+	}
+
+	return files[idx:], files[idx:]
```
Comment (3pointer) [inline pkg/restore/client.go:573]: files[:idx], files[idx:] 🤔️？

Comment (glorv) [reply pkg/restore/client.go]: 😂

**Pull request merged**